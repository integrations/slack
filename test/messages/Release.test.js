const { Release } = require('../../lib/messages/release');
const repository = require('../fixtures/repo.json');
const release = require('../fixtures/release.json');

describe('Release rendering', () => {
  test('works', async () => {
    const releaseMessage = new Release({
      eventType: 'release.published',
      release,
      repository,
    });
    const rendered = releaseMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
  test('works for empty release name', async () => {
    const r = { ...release };
    r.name = '';
    const releaseMessage = new Release({
      eventType: 'release.published',
      release: r,
      repository,
    });
    const rendered = releaseMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
  test('works for null release name', async () => {
    const r = { ...release };
    r.name = null;
    const releaseMessage = new Release({
      eventType: 'release.published',
      release: r,
      repository,
    });
    const rendered = releaseMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
});
