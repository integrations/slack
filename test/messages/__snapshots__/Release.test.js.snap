// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release rendering works 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#24292f",
  "fallback": "[bkeepers/dotenv] Release - vX.Y.Z - Canary Dwarf",
  "footer": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
  "footer_icon": "https://assets-cdn.github.com/favicon.ico",
  "image_url": undefined,
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "*Features*

• Feature 1  
    ...

*Fixes*

• Fix 1  
    ...

*Internals*

• Internal improvement 1  
    ...",
  "title": "Release - vX.Y.Z - Canary Dwarf",
  "title_link": "https://github.com/github-slack/app/releases/tag/vX.Y.Z",
  "ts": 1523863209,
}
`;

exports[`Release rendering works for empty release name 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#24292f",
  "fallback": "[bkeepers/dotenv] Release - vX.Y.Z",
  "footer": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
  "footer_icon": "https://assets-cdn.github.com/favicon.ico",
  "image_url": undefined,
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "*Features*

• Feature 1  
    ...

*Fixes*

• Fix 1  
    ...

*Internals*

• Internal improvement 1  
    ...",
  "title": "Release - vX.Y.Z",
  "title_link": "https://github.com/github-slack/app/releases/tag/vX.Y.Z",
  "ts": 1523863209,
}
`;

exports[`Release rendering works for null release name 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#24292f",
  "fallback": "[bkeepers/dotenv] Release - vX.Y.Z",
  "footer": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
  "footer_icon": "https://assets-cdn.github.com/favicon.ico",
  "image_url": undefined,
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "*Features*

• Feature 1  
    ...

*Fixes*

• Fix 1  
    ...

*Internals*

• Internal improvement 1  
    ...",
  "title": "Release - vX.Y.Z",
  "title_link": "https://github.com/github-slack/app/releases/tag/vX.Y.Z",
  "ts": 1523863209,
}
`;
